lib/backend.js

```diff
@@ -661,25 +661,59 @@ Backend.prototype.querySubscribe = function(agent, index, query, options, callba
         'DB does not support subscribe'
       ));
     }
-    backend.pubsub.subscribe(request.channel, function(err, stream) {
-      if (err) return callback(err);
+
+    var channels = request.channels;
+
+    if (request.channel) {
+      logger.warn('[DEPRECATED] Do no use channel for query, use channels instead.');
+      channels = [request.channel];
+    }
+
+    if (!channels || !channels.length) {
+      return callback(new Error('Required minimum one query channel.'));
+    }
+
+    var streams = [];
+
+    function cleanStreams() {
+      streams.forEach(function(stream) {
+        stream.destroy();
+      });
```

Comment:
We can consider clearing `streams` at the end, in case this function is called multiple times. Alec suggests setting `streams.length` to 0.